### PR TITLE
Update 3.0 release notes

### PR DESCRIPTION
**Pull request by waylan**
And other docs cleanup.

### CODE REVIEW

**Comment by waylan** [comment]
@mitya57 @facelessuser I [added][1] both of you to the list of "authors". If you would like anything changed (for example, I linked to your GitHub profile, you may prefer a personal website), please make the changes. You should just be able to use GitHub's web site to `edit` the file within the PR.

There is a lot of stuff in the release notes, so a few extra reviewers would be helpful. Is their anything I missed? Anything that is not clear? Anything that could be simplified?

[1]: https://github.com/Python-Markdown/markdown/pull/698/files#diff-365b0403c12f88b9d3bbdfb3f3b077a4

File: docs/change_log/release-3.0.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+If you are not interested in sanitizing untrusted text, but simply desire to
+escape raw HTML, then that can be accomplished through an extension which
+removes HTML parsing:
+
+```python
+from markdown.extensions import Extension
+
+class EscapeHtml(Extension):
+    def extendMarkdown(self, md, md_globals):
```
Comment (mitya57) [inline docs/change_log/release-3.0.md]: Maybe remove `md_globals` now that it’s deprecated?

File: docs/change_log/release-3.0.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+The old attributes on class instances still exist, but raise a
+`DeprecationWarning` when accessed. Also on classes where the instance was
+optional, the attribute always exists now and is simply `None` if no instance
+was provided (previously the attribute would not exist).
+
+### `markdown.util.isBLockLevel` deprecated
+
+The `markdown.util.isBLockLevel` function is deprecated and will raise a
+`DeprecationWarning`. Instead, extensions should use the `isBLockLevel` method
```
Comment (mitya57) [inline docs/change_log/release-3.0.md]: In these three lines the capitalization of `BLock` looks wrong.

File: docs/change_log/release-3.0.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+A new `InlineProcessor` class handles inline processing much better and allows
+for more flexibility. The new `InlineProcessor` classes no longer utilize
+unnecessary pretext and post-text captures. New class can accept the buffer that
+is being worked on and manually process the text without regular expressions and
+return new replacement bounds. This helps us to handle links in a better way and
+handle nested brackets and logic that is too much for regular expression.
+
+## New features
+
+The following new features have included in the release:
```
Comment (mitya57) [inline docs/change_log/release-3.0.md]: have → have been?

File: docs/change_log/release-3.0.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+* A new [testing framework](../test_tools.md) is included as a part of the
+  Markdown library, which can also be be used by third party extensions.
```
Comment (mitya57) [inline docs/change_log/release-3.0.md]: be be → be

**Comment by facelessuser** [comment]
Thanks @waylan, I will try to get to this by at least this weekend.

**Review by mitya57 — CHANGES_REQUESTED** [review]
Thanks for listing me as an author :)
The changes look good, just some minor comments.

File: docs/change_log/release-3.0.md (hunk 1 above)
Comment (waylan) [reply docs/change_log/release-3.0.md]: fixed.

File: docs/change_log/release-3.0.md (hunk 2 above)
Comment (waylan) [reply docs/change_log/release-3.0.md]: fixed

File: docs/change_log/release-3.0.md (hunk 3 above)
Comment (waylan) [reply docs/change_log/release-3.0.md]: fixed

File: docs/change_log/release-3.0.md (hunk 4 above)
Comment (waylan) [reply docs/change_log/release-3.0.md]: fixed

**Comment by waylan** [comment]
Thanks for the review @mitya57. Feel free to mark my changes as "accepted" if they address the concerns raised.

In any event, everything is starting to blur together, I need to step away and take a fresh look later. I will be coming back to do a review in the next few days.

**Review by mitya57 — APPROVED** [review]

File: docs/extensions/api.md [hunk 5]
```diff
@@ -554,109 +548,75 @@ A simple example:
...
-proper order. At the same time, you can use `insert()`, `append()`, and
-`index()` as you would with a list.
-
-Generally speaking, within Markdown extensions you will be using the special
-helper method `add()` to add additional items to an existing OrderedDict.
+### Registry
 
-The `add()` method accepts three arguments:
+The `markdown.util.Registry` class is a priority sorted registry which Markdown
+uses internally to determine the processing order if its various processors and
```
Comment (facelessuser) [inline docs/extensions/api.md]: order **of** its various processors

File: docs/extensions/api.md [hunk 6]
```diff
@@ -554,109 +548,75 @@ A simple example:
...
+registry: `register` and `deregister`. Use `register` to add items and
+`deregister` to remove items. See each method for specifics.
 
-* **`value`**: The object instance stored in this item.
+When registering an item, a "name" and a "priority" must be provided. All
+items are automatically sorted by "priority" from highest to lowest. The
+"name" is used to remove (`deregister`) and get items.
 
-* **`location`**: The items location in relation to other items.
+A `Registry` instance it like a list (which maintains order) when reading
```
Comment (facelessuser) [inline docs/extensions/api.md]: instance **is** like a list

**Review by facelessuser — CHANGES_REQUESTED** [review]

**Comment by facelessuser** [comment]
I found a couple of typos that need fixing, but I was able to get through the documentation today, so I'm done.

**Comment by waylan** [comment]
@facelessuser please mark my changes as "accepted" if they address the concerns raised. Thank you.

**Review by facelessuser — APPROVED** [review]

**Pull request merged**